processor/src/main/java/org/mapstruct/ap/internal/model/AbstractMappingMethodBuilder.java

```diff
@@ -73,7 +73,7 @@ Assignment forgeMapping(SourceRHS sourceRHS, Type sourceType, Type targetType) {
             true
         );
 
-        return createForgedBeanAssignment( sourceRHS, forgedMethod );
+        return createForgedAssignment( sourceRHS, forgedMethod );
```

Comment:
I was thinking the same @sjaakd. That is what I meant with:
>Maybe we need to do the same that it is done in MapperCreationProccessor.getMappingMethods()

If you all agree, I can theoretically do it as part of this PR, or you prefer a new issue?